### ISSUE TITLE
Duration in hours

### ISSUE THREAD

**Issue by GoogleCodeExporter** [bardsoftware/ganttproject#225]
```
What steps will reproduce the problem?
1. When adding a task, it is assumed that the duration is one.  It isn't 
possible to reduce this below 1.  The measurement is in days - what if there is 
a half day or 3 hour task?

It would be good to be able to indicate 1hr, 3hrs or 7hrs or 1 day.  I guess 
Hours, Days, Months would be the desired units.


What version of the product are you using? On what operating system?
Using 2.0.10 on Windows 7.

```

Original issue reported on code.google.com by `Harborou...@gmail.com` on 19 Apr 2011 at 2:38

**Comment by RadialAdmin** [comment]
I update the issue, to recall that it is really important to many users.

**Comment by ravelus** [comment]
Cannot believe this is not implemented yet after *years* of divagation... this is a must!!!

**Comment by griznold** [comment]
Is there any update? I would really appreciate this feature, too.

**Comment by GRA0007** [comment]
Yes it would be good to have this feature. At the moment I have to use excel for projects that use minutes or hours. Does anyone have an update on the progress of this issue?

**Comment by Hillside502** [comment]
The duration less than a day (MISSING) "feature" is a total and utter dealbreaker.

It has been debated/requested/IGNORED for over 12 years, as follows:-

GanttProject / Feature Requests / #34 Hour/Minutes/Seconds granularity of task duration
https://sourceforge.net/p/ganttproject/feature-requests/34/

Currently, this follows on at:-

Duration in hours · Issue #225 · bardsoftware/ganttproject
https://github.com/bardsoftware/ganttproject/issues/225

Why on earth anyone/everyone would NOT want tasks with a duration of LESS THAN A DAY is beyond me.

Bye, Bye GanttProject --- dead in the water --- before I give it any more of my "time" and "effort"!

**Comment by sashimiFR** [comment]
I'm discovering Ganttproject, and immediately ran into this duration granularity issue too. Be it dealing with professional projects in my consulting work, or trying to optimise how I'm going to do the 3 layers of painting of each 2 sides of the 12 shutters of my home, adequate timescaling is brick one of any project management with a chance of success... It can't and shouldn't be a random "feature" which can be delayed for over 10 years. And being free software is no excuse. If you're going to build a project management app, do it properly or don't bother doing it at all (I mean, really, what's the point ?).

Begs me to ask the dev : you earlier yourself mentioned you work on ganttproject 1h/day or so. So HH:MM:SS timescale makes sense even for your own sake. Or do you don't use your very own project management tool to manage your own projects ??! (which would explain quite a bit).

Ganttproject is broken by design ; what a great pity.

**Comment by marcomavil** [comment]
I would gladly help improve this program by addin' the granularity of task duration feature (really without it this program is pretty much useless) I understand how this kind of programs work and calculate progress, effort, performance, etc, since I've been working under Oracle Primavera for years...

The thing is I'm new to github (It's my first day, LOL) and I don't know how can I help improve GanttProject...

**Comment by dbarashev** [comment]
@marcomavil this feature is one of the most difficult to implement and basically requires major rewrite of nearly half of the code base. If you're up for some open-source coding, I would really recommend to start with more simple features. Please contact us by email if interested.

**Comment by netroby** [comment]
Getting the time min range to per hour(not per day ) will help us a lot.

**Comment by drwerewolf** [comment]
I agree with the comment above.

**Comment by Hillside502** [comment]
Nearly 3 years later --- a tosser by any other name!

**Comment by dbarashev** [comment]
I am locking the conversation because just writing "it is a must", "I need it too", "so many years later", "bye-bye", etc. is not so much helpful. 

This feature requires major rewriting of nearly half of the code base. It can be done by a very high skilled engineer in roughly a few weeks, and costs a much as a few weeks of such engineer cost, that is, a few thousands of dollars/euros. If you are willing to pay that much, let us know by email. If you are a star software engineer who is proficient with algorithms and Java//Swing/JavaFX, and you are willing to donate your time, please contact us as well. Even if working for free is not your cup of tea, but you feel enthusiastic and have enough skills, and willing to work for less than you would get in Google, you may want to contact us as well.

Otherwise, just take it that at the moment being GanttProject is not the best fit for your needs.
